nexus/src/app/instance.rs

```diff
@@ -563,9 +563,12 @@ impl super::Nexus {
         // outright fails, this operation fails. If the operation nominally
         // succeeds but nothing was updated, this action is outdated and the
         // caller should not proceed with migration.
-        let (updated, _) = match instance_put_result {
+        let (updated, _, _) = match instance_put_result {
             Ok(state) => {
                 self.write_returned_instance_state(&instance_id, state).await?
+
+                // TODO(eliza): this is where we would also want to insert to the
```

Comment:
Is it a problem we aren't doing so in this PR?

(I'm okay splitting things out, just want to make sure we don't cause regressions by leaving this as a comment)

Reply:
^ this is basically my only blocking comment, if we're okay with this, the whole PR LGTM

Reply:
I explained this in response to one of Greg's comments: the comment was just stale, I've removed it: https://github.com/oxidecomputer/omicron/pull/5859#discussion_r1633845748